src/api/github-client.ts

```diff
@@ -0,0 +1,239 @@
+/*********************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import { Octokit } from '@octokit/rest'
+
+const OWNER = 'eclipse'
+export const CHE_REPO = 'che'
+export const CHE_OPERATOR_REPO = 'che-operator'
+
+export interface TagInfo {
+  name: string
+  commit: {
+    sha: string
+    url: string
+  }
+  zipball_url: string
+}
+
+export class CheGithubClient {
+  private readonly octokit: Octokit
+
+  constructor() {
+    this.octokit = new Octokit({
+      baseUrl: 'https://api.github.com',
+      userAgent: 'chectl',
+      auth: process.env.GITHUB_TOKEN,
```

Comment:
what happens if I don't have any GITHUB_TOKEN env variable ? (almost all users)

Reply:
Nothing critical. It will usually work. The token was added just for case, for example when some requests limit is hit. But while I was developing/debugging the code I did a lot of requests and never hit the limit (if any for such kind of requests). So I think it safe to use it without token, but still leaving a posibility to authenticate.